lib/elixir/lib/task.ex

```diff
@@ -752,18 +789,29 @@ defmodule Task do
   a response waiting in the message queue this function will return
   `{:exit, :noproc}` as the result or exit reason can not be determined.
   """
-  @spec shutdown(t, timeout | :brutal_kill) :: {:ok, term} | {:exit, term} | nil
-  def shutdown(task, shutdown \\ 5000)
+  @spec shutdown(t, timeout :: non_neg_integer | :brutal_kill) ::
+          {:ok, term} | {:exit, term} | nil
+  def shutdown(task, timeout_or_brutal_kill \\ 5000)
 
-  def shutdown(%Task{pid: nil} = task, _) do
-    raise ArgumentError, "task #{inspect(task)} does not have an associated task process"
-  end
+  def shutdown(%Task{pid: pid, owner: owner} = task, timeout)
+      when is_timeout(timeout) or timeout == :brutal_kill do
+    cond do
+      timeout == :infinity ->
+        raise ArgumentError,
+              ":infinity is not a valid timeout, please use a value in milliseconds or :brutal_kill"
 
-  def shutdown(%Task{owner: owner} = task, _) when owner != self() do
-    raise ArgumentError, invalid_owner_error(task)
+      is_nil(pid) ->
+        raise ArgumentError, "task #{inspect(task)} does not have an associated task process"
+
+      owner != self() ->
+        raise ArgumentError, invalid_owner_error(task)
```

Comment:
Shutdown does not require this check because we always monitor ourselves (to avoid data races).